src/Configuration/ConfigurationResolver.php

```diff
@@ -100,6 +114,20 @@ public static function ini($key, $expectedType, $profile = null, $filename = nul
         // Use INI_SCANNER_NORMAL instead of INI_SCANNER_TYPED for PHP 5.5 compatibility
         //TODO change after deprecation
         $data = @\Aws\parse_ini_file($filename, true, INI_SCANNER_NORMAL);
+
+        if (isset($options['section'])
+            && isset($options['subsection'])
+            && isset($options['key']))
+        {
+            return self::retrieveValueFromIniSubsection(
```

Comment:
As this is expected to grow and add more config options to individual services, does it make more sense to grab the subsection early on in the client resolver instead and reference it here and have that be referenced throughout the resolution of the client?